### PR TITLE
Prevent `Content-Type` override if `Content-Length` not set

### PR DESCRIPTION
**Pull request by coditva**
## PR Checklist:
- [x] I have run `npm test` locally and all tests are passing.
- [x] I have added/updated tests for any new behavior.
       
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: https://github.com/postmanlabs/postman-app-support/issues/1392
       

## PR Description

`Content-Type` was re-written if `Content-Length` was not present. This change prevents that behaviour. Also adds test for specifying custom charset in the `Content-Type` header.

### CODE REVIEW

File: tests/test-form-data-boundary.js [hunk 1]
```diff
@@ -127,6 +125,29 @@ tape('custom boundary with content-length', function (t) {
   })
 })
 
+tape('custom boundary and charset', function (t) {
+  var boundary = '----WebKitFormBoundary7MA4YWxkTrZu0gW'
```
Comment (codenirvana) [inline tests/test-form-data-boundary.js]:
```suggestion
  var boundary = 'X-FORM-DATA-BOUNDARY'
```

File: tests/test-form-data-boundary.js [hunk 2]
```diff
@@ -105,8 +105,6 @@ tape('custom boundary with content-length', function (t) {
   request.post({
     url: server.url,
     headers: {
-      // @note `content-type` header is not re-calculated if `content-length`
-      // is present for formData body
       'content-type': 'multipart/anything; boundary=' + boundary,
```
Comment (codenirvana) [inline tests/test-form-data-boundary.js]: Add a similar test without `content-type`, it should generate content-type correctly.

Comment (coditva) [reply tests/test-form-data-boundary.js]: Added in b4003e0

File: tests/test-form-data-boundary.js [hunk 3]
```diff
@@ -100,13 +100,35 @@ tape('custom boundary within quotes', function (t) {
...
+  }, function (err, res, body) {
+    var req = JSON.parse(body)
+    var boundary
+    t.equal(err, null)
+    t.equal(res.statusCode, 200)
+    t.notEqual(req.headers['content-type'], null)
+    boundary = req.headers['content-type'].split('boundary=')[1]
+    t.equal(req.headers['content-type'], 'multipart/form-data; boundary=' + boundary)
+    t.equal(req.headers['content-length'], '171')
+    t.ok(req.body.indexOf('name="formKey"') !== -1)
```
Comment (codenirvana) [inline tests/test-form-data-boundary.js]: Also test for boundary on `req.body`, like its done in other tests:
```js
t.ok(req.body.startsWith('--' + boundary))
t.ok(req.body.endsWith(boundary + '--\r\n'))
```

Comment (coditva) [reply tests/test-form-data-boundary.js]: Added in c9cfe9f
